app/src/main/java/org/oppia/android/app/player/state/StatePlayerRecyclerViewAssembler.kt

```diff
@@ -1183,6 +1183,17 @@ class StatePlayerRecyclerViewAssembler private constructor(
       }
     }
 
+    /**
+     * Add AdapterBuilderFactory passed through injection from [StateFragmentPresenter]
+     */
+    fun addAdapterBuilderFactory
+    (multiTypeBuilderFactory: BindableAdapter.MultiTypeBuilder.Factory): Builder {
+      adapterBuilder = multiTypeBuilderFactory.create {
+        it.viewType
+      }
+      return this
+    }
```

Comment:
Do we actually need this? I think you can inject the factory in the assembler's own injectable factory to avoid needing the extra plumbing.

Reply:
Have refactored this too